source/game/field/ObjectCollisionBase.cc

```diff
@@ -1,9 +1,323 @@
+    }
+
+    v /= sum;
+}
+
+/// @addr{0x80835A8C}
+void ObjectCollisionBase::calcSimplex(GJKState &state) const {
+    const u32 idx = state.m_idx;
+
+    for (u32 i = 0, mask = 1; i < 4; ++i, mask *= 2) {
+        if ((state.m_flags & mask) == 0) {
+            continue;
+        }
+
+        f32 result = state.m_s[i].dot(state.m_s[idx]);
+        CACHE[idx][i] = result;
+        CACHE[i][idx] = result;
+    }
+
+    state.m_scales[state.m_mask][idx] = 1.0f;
+    CACHE[idx][idx] = state.m_s[idx].dot();
+
+    for (u32 i = 0, iMask = 1; i < 4; ++i, iMask *= 2) {
+        if ((state.m_flags & iMask) == 0) {
+            continue;
+        }
+
+        u32 iStateMask = iMask | state.m_mask;
+
+        state.m_scales[iStateMask][i] = CACHE[idx][idx] - CACHE[idx][i];
+        state.m_scales[iStateMask][idx] = CACHE[i][i] - CACHE[i][idx];
+
+        for (u32 j = 0, jMask = 1; j < i; ++j, jMask *= 2) {
+            if ((state.m_flags & jMask) == 0) {
+                continue;
+            }
+
+            state.m_scales[jMask | iStateMask][j] =
+                    state.m_scales[iStateMask][i] * (CACHE[i][i] - CACHE[i][j]) +
+                    state.m_scales[iStateMask][idx] * (CACHE[idx][i] - CACHE[idx][j]);
+            state.m_scales[jMask | iStateMask][i] =
+                    state.m_scales[jMask | state.m_mask][j] * (CACHE[j][j] - CACHE[i][j]) +
+                    state.m_scales[jMask | state.m_mask][idx] * (CACHE[idx][j] - CACHE[idx][i]);
+            state.m_scales[jMask | iStateMask][idx] =
+                    state.m_scales[jMask | iMask][j] * (CACHE[j][j] - CACHE[j][idx]) +
+                    state.m_scales[jMask | iMask][i] * (CACHE[i][j] - CACHE[i][idx]);
+        }
+    }
+
+    if (state.m_00c != 0xf) {
+        return;
+    }
+
+    const f32 _1_1 = CACHE[0][0] - CACHE[0][1];
+    const f32 _2_2 = CACHE[0][0] - CACHE[0][2];
+    const f32 _3_2 = CACHE[0][0] - CACHE[0][3];
+    const f32 _0_2 = CACHE[1][1] - CACHE[1][0];
+    const f32 _0_3 = CACHE[2][1] - CACHE[2][0];
+    const f32 _1_2 = CACHE[3][0] - CACHE[3][1];
+    const f32 _1_3 = CACHE[2][0] - CACHE[2][1];
+    const f32 _2_1 = CACHE[3][0] - CACHE[3][2];
+    const f32 _0_1 = CACHE[3][1] - CACHE[3][0];
+    const f32 _2_3 = CACHE[1][0] - CACHE[1][2];
+    const f32 _3_1 = CACHE[2][0] - CACHE[2][3];
+    const f32 _3_3 = CACHE[1][0] - CACHE[1][3];
+
+    state.m_scales[15][0] = _0_1 * state.m_scales[14][3] +
+            (_0_2 * state.m_scales[14][1] + _0_3 * state.m_scales[14][2]);
+    state.m_scales[15][1] = _1_2 * state.m_scales[13][3] +
+            (_1_1 * state.m_scales[13][0] + _1_3 * state.m_scales[13][2]);
+    state.m_scales[15][2] = _2_1 * state.m_scales[11][3] +
+            (_2_2 * state.m_scales[11][0] + _2_3 * state.m_scales[11][1]);
+    state.m_scales[15][3] = _3_1 * state.m_scales[7][2] +
+            (_3_2 * state.m_scales[7][0] + _3_3 * state.m_scales[7][1]);
+}
+
+std::array<std::array<f32, 4>, 4> ObjectCollisionBase::CACHE = {{}};
```

Comment:
This is a static member of a class - should have an `s_` prefix.

Can we find a more descriptive name other than CACHE? If we can't, it's fine.

Reply:
Let's call this `S_DOT_PRODUCT_CACHE`